lib/rules/destructuring-assignment.js

```diff
@@ -22,12 +22,20 @@ module.exports = {
         'always',
         'never'
       ]
+    }, {
+      type: 'object',
+      properties: {
+        ignoreClassFields: {
+          type: 'boolean'
+        }
+      },
+      additionalProperties: false
     }]
   },
 
   create: Components.detect((context, components, utils) => {
     const configuration = context.options[0] || DEFAULT_OPTION;
-
+    const ignoreClassFields = configuration === 'always' && context.options[1] && context.options[1].ignoreClassFields === true || false;
```

Comment:
when set to “never”, this should still apply - you could maybe have `foo = function () { const { a } = this.props; }.call(this)`, or when do expressions land, you could have `foo = do { const { a } = this.props; a; }`

Reply:
Makes sense, I'll update my PR 👍

Reply:
shouldn't https://github.com/yannickcr/eslint-plugin-react/pull/1909/files#diff-d73ce54079fb0715f60db27176a09fb6R135 also be updated?

Reply:
Yep it probably should. Sorry for missing that, I've never worked in AST or eslint-plugin projects before so it's all new to me. I've updated the PR again.